components/Habit.tsx

```diff
@@ -62,17 +70,35 @@ type HabitProps = {
   streak: number;
   habit: THabit;
   entries: Entry[];
+  addHabitEntry: () => void;
+  removeHabitEntry: () => void;
 };
 
-const Habit = ({ entries, habit, streak }: HabitProps) => {
+const Habit = ({
+  entries,
+  habit,
+  streak,
+  addHabitEntry,
+  removeHabitEntry,
+}: HabitProps) => {
   const { t } = useTranslation(['common', 'habit']);
 
-  const todayEntries = getEntriesForToday(entries);
+  const [todayEntries, setTodayEntries] = useState<Entry[]>([]);
+
   const completionPercentage = getCompletionPercentage(
     checkCompletionQuantity(todayEntries),
     habit.target.quantity
   );
 
+  const toggleComplete = () => {
+    if (todayEntries.length >= 1) removeHabitEntry();
+    else addHabitEntry();
+  };
+
+  useEffect(() => {
+    setTodayEntries(getEntriesForToday(entries));
+  }, [entries]);
```

Comment:
I did this because I thought that when the entries value changed, it wasn't re-rendering and updating properly. Doing this fixed the issue I mentioned to you before about a habit not being able to be untoggled after marking it as completed.

Reply:
Believe this is resolved in Slack, correct me if wrong

Reply:
Thanks. I changed it.